hdrop/hdrop

```diff
@@ -1,83 +1,106 @@
+  echo "  [COMMAND]"
+  echo "          The usual command you would run to start the desired program"
   echo ""
-  echo "Several instances of the same program can be run concurrently, if different class names are assigned to each instance. Presently there is support for the following flags:"
+  echo "Options:"
+  echo "  -b, --background"
+  echo "          changes the default behaviour: if the specified program is"
+  echo "          not running, launch it in the background instead of foreground."
+  echo "          Thereafter 'hdrop -b' will work the same as without this flag."
   echo ""
-  echo " -a ('foot' terminal emulator)"
-  echo " --class (all other programs)."
+  echo "  -h, --help"
+  echo "          Print this help message"
   echo ""
-  echo "Please feel free to report missing flags for other programs!"
+  echo "  -V, --version"
+  echo "          Print version"
   echo ""
-  echo "Example bindings in Hyprland config:"
+  echo "Multiple instances:"
   echo ""
-  echo ' bind = $mainMod, b, exec, hdrop librewolf'
-  echo ' bind = $mainMod, x, exec, hdrop kitty --class kitty_1'
-  echo ' bind = $mainMod CTRL, x, exec, hdrop kitty --class kitty_2'
-  echo ' bind = $mainMod, c, exec, hdrop foot -a foot_1'
-  echo ' bind = $mainMod CTRL, c, exec, hdrop foot -a foot_2'
+  echo "Multiple instances of the same program can be run concurrently, if"
+  echo "different class names are assigned to each instance. Presently there is"
+  echo "support for the following flags in the [COMMAND] string:"
   echo ""
-  echo "Note: defining a class name is only necessary when running several instances of the same program."
-  echo ""
-  echo "If you want to run a program on boot and have it wait in the background until called up by hdrop you can use this:"
-  echo ""
-  echo " exec-once = [workspace special:hdrop silent] librewolf"
-  echo ""
-  echo "Dependencies: bash jq hyprland"
-  echo "Part of: https://github.com/hyprwm/contrib"
-  echo "Author: Schweber (https://github.com/Schweber/hdrop)"
-  echo "License: AGPL3"
+  echo " -a         ('foot' terminal emulator)"
+  echo " --class    (all other programs)."
   echo ""
+  echo "See man page for more information"
+}
+
+print_version() {
+  echo "hdrop version: 0.2.1"
+}
+
+notify() {
+  notify-send "$@" || hyprctl notify 0 5000 0 "$@"
+}
+
+BACKGROUND=""
+
+case "$1" in
+"")
+  notify "hdrop: missing Argument" "Run 'hdrop -h' for more information"
+  print_help
+  exit 1
+  ;;
+-h | --help)
+  print_help
   exit
   ;;
-"foot")
-  TEMP=$(getopt --options a: -n hdrop -- "$@")
+-V | --version)
+  print_version
+  exit
   ;;
-*)
-  TEMP=$(getopt --longoptions class: -n hdrop -- "$@")
+-b | --background)
+  BACKGROUND="hyprctl dispatch -- exec [workspace special:hdrop silent] "
+  shift
+  ;;
+-*)
+  notify "hdrop: unrecognized flag" "Run 'hdrop -h' for more information"
+  print_help
+  exit 1
   ;;
 esac
 
-eval set -- "$TEMP"
+ACTIVE_WORKSPACE="$(hyprctl activeworkspace -j | jq -r .id)"
```

Comment:
I know it's not the first time `jq` is being used in this script, but perhaps we should copy the `check` functionality from `grimblast`, to notify a user whether they've missed a dependency.

Reply:
I had it in there initially (among other dependency checks) but removed them with the previous merge because `hyperfine` showed me around 42ms of execution time. With everything ripped out it's below 9ms now so i'd like to keep it that way.

I now added a solution that shouldn't cost execution time if no errors occur.